### ISSUE TITLE
How to - I want use privatebin for copy a log file from console to my privatebin

### ISSUE THREAD

**Issue by ZerooCool** [PrivateBin/PrivateBin#649]
How to - I want use privatebin for copy a log file from console to my privatebin

**Comment by ZerooCool** [comment]
How to ? PBinCLI
https://wiki.visionduweb.fr/index.php?title=Exporter_un_fichier_texte_vers_un_service_en_ligne_de_type_pastebin#PBinCLI

How to ? Cpaste
https://wiki.visionduweb.fr/index.php?title=Exporter_un_fichier_texte_vers_un_service_en_ligne_de_type_pastebin#Cpaste

How to ? PrivateBin-Groovy
https://wiki.visionduweb.fr/index.php?title=Exporter_un_fichier_texte_vers_un_service_en_ligne_de_type_pastebin#PrivateBin-Groovy

Work ! Privatebin-cli
https://wiki.visionduweb.fr/index.php?title=Exporter_un_fichier_texte_vers_un_service_en_ligne_de_type_pastebin#Privatebin-cli

**Comment by ZerooCool** [comment]
Work ! PBinCLI
https://wiki.visionduweb.fr/index.php?title=Exporter_un_fichier_texte_vers_un_service_en_ligne_de_type_pastebin#PBinCLI

Work ! Privatebin-cli
https://wiki.visionduweb.fr/index.php?title=Exporter_un_fichier_texte_vers_un_service_en_ligne_de_type_pastebin#Privatebin-cli

**Comment by ZerooCool** [comment]
You can add my link, or, a translation, for PBinCLI and Privatebin-cli to the documentation for use a client with PrivateBin.

Can you help for check :
Cpaste
PrivateBin-Groovy

**Comment by ZerooCool** [comment]
How to install and use ? Cpaste
https://wiki.visionduweb.fr/index.php?title=Exporter_un_fichier_texte_vers_un_service_en_ligne_de_type_pastebin#Cpaste

For Cpaste, I only understood how to send to the privatebin, a text, or, a file, read with echo or cat.
To pass the options, I do not understand.

`echo OK | python Cpaste.py`

**Comment by ZerooCool** [comment]
How to install and use ? PrivateBin-Groovy
https://wiki.visionduweb.fr/index.php?title=Exporter_un_fichier_texte_vers_un_service_en_ligne_de_type_pastebin#PrivateBin-Groovy

**Comment by elrido** [comment]
cpaste: On top of the file it describes the libraries that would need to be installed via pip or other package manager. I does have a help option, so `./cpaste --help` should give you some hints on what flags need to be set.

PrivateBin-Groovy: I see that you opened an [issue with the upstream project](https://github.com/kkingsley-BF/PrivateBin-Groovy/issues/1), perfect!

**Comment by ZerooCool** [comment]
@elrido 
With cpaste, i have only the script.py ( then, i can say : cpaste.py )
Then, i need test with ./cpaste.py --help ?
Ok, i test that, ASAP.

Yes for Groovy, It is the only one that I did not use, since finally, I managed to make work the 3 others, and, to also go up some errors.

I would keep the issue open while I update the documentation for Cpaste and Groovy.
The shared links to my wiki allow copy and paste, all stages of client installation are mentioned, plus examples for writing text or the content of a text file to privatebin.

**Comment by ZerooCool** [comment]
#188

**Comment by ZerooCool** [comment]
Ok for Cpaste, client work.

I don't know how to pass the options after ordering, I have to use echo and pipe.

```
python ./cpaste.py --help
Usage: cpaste.py [options]

Options:
  -h, --help            show this help message and exit
  -f FILE, --file=FILE  Read from a file instead of stdin
  -p PASSWORD, --password=PASSWORD
                        Create a password protected paste
  -e EXPIRE, --expire=EXPIRE
                        Expiration time of the paste (default: 1day)
  -s, --sourcecode      Use source code highlighting
  -m, --markdown        Parse paste as markdown
  -b, --burn            Burn paste after reading
  -o, --opendiscussion  Allow discussion for the paste
  -a FILE, --attachment=FILE
                        Specify path to a file to attachment to the paste
```

# Afficher un texte simple :
`echo OK | python cpaste.py`

# Afficher un texte en Markdown :
`echo "# Hello World" | python cpaste.py -m`

**Comment by elrido** [comment]
If I understand the above help text correctly, you can pass your log file either with `-f` to be displayed inline or using `-a` to send as a downloadable attachment.

**Comment by ZerooCool** [comment]
But, for a simple text, i can't, i don't know how i can make that.
Then, i use " echo texte | " or cat " text-file | "

**Comment by elrido** [comment]
I think that is exactly how you should use it. At least it claims to read from `stdin` by default.

**Comment by rugk** [comment]
All third-party clients are listed on https://github.com/PrivateBin/PrivateBin/wiki/Third-party-clients. If you find more, feel free to add them there.

So as it seems this question has been answered and I thus close this issue. If anyone has further questions on that topic, feel free to comment here, again.

**Comment by ZerooCool** [comment]
I actually followed this link, to discover the 4 privatebin clients.
The last client, Groovy, has poor documentation, it seems to me, I could be wrong, but the implementation was not fast.

This is why I made a proposal, on my wiki, to use (faster) privatebin clients.

The examples are functional, in French.
https://wiki.visionduweb.fr/index.php?title=Exporter_un_fichier_texte_vers_un_service_en_ligne_de_type_pastebin#Clients_Privatebin

I would have liked to have had a return, for Groovy, but, for the moment, I have not had one.
